ironfish/src/network/peerNetwork.ts

```diff
@@ -311,6 +311,27 @@ export class PeerNetwork {
     this.peerManager.broadcast(message)
   }
 
+  /**
+   * Send a block to all connected peers who haven't yet received the block.
+   */
+  private broadcastBlock(message: NewBlockMessage): void {
+    this.seenGossipFilter.add(message.nonce)
+
+    // TODO: This deserialization could be avoided by passing around a Block instead of a SerializedBlock
+    const block = this.strategy.blockSerde.deserialize(message.block)
+
+    for (const peer of this.peerManager.getConnectedPeers()) {
+      // Don't send the block to peers who already know about it
+      if (peer.knownBlockHashes.has(block.header.hash)) {
+        continue
+      }
+
+      if (peer.send(message)) {
+        peer.knownBlockHashes.set(block.header.hash, false)
```

Comment:
I know we don't technically use this but you use false here, and true in another place. Though if it is true, and someone forgets to use `has()` it'll still work.

Reply:
aaah, I see. In that case it makes sense. I would ignore my comment them. If you want to create an LRUSet I think that would be fine, but otherwise this seems resolved.

Reply:
I would agree that it's misleading at first glance that `peer.knownBlockHashes.get(x)` returning false would mean it IS in the set, but it was received vs sent.

For a more memory-friendly but verbose implementation maybe could use an enum that has states like `RECEIVED` and `SENT`?

Reply:
Oh, I like that idea -- let me see if I can make that work with the LRUMap.